sphinx circle integration

bentiss opened the pull request:
PR on top of your PR :)

This are the circle.yml changes to check the building of the docs and the automation of the generation of libratbag.github.io.

whot commented:
merged, thanks!

whot commented:
just fyi, I rebased this on top of github.com/libratbag/libratbag, so don't get too confused when the next merge isn't fast-forward.

whot committed:
Remove c++ guard

libratbag is not a public header, so we don't have to worry about C++ anymore.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

whot committed:
Move the enums into their separate header file

libratbag is not a library anymore. Move all the stuff that we need for the
DBus interface (i.e. the enum values) into its own header file so they're
all in one place and we don't have to worry about it anymore.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

phomes committed:
hidpp10: Fix LED read/write

With this the LEDs can be read and written on the G9.

The G9 is as far as I know the only hid++1.0 mouse with RGB LED. The
other mice are monochrome.

whot committed:
Use Sphinx for the DBus interface documentation

Better than the markdown we were using before, even though we're currently
(ab)using the js/c++ domains for the structure. There is no DBus domain in
sphinx yet.

The main motivation was to get the enums into the documentation since they are
exported ABI.

Generating the documentation is quirky - we use doxygen to extract the
comments from the C header file and then use XSLT to convert to .rst which
sphinx can use. Still better than having to maintain even more copies of the
enums.

Since this is only of interest for maintainers that need to upload the
documentation, it is turned off by default.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

bentiss committed:
circle.yml: deploy docs when the master branch is updated

Note that the push requires a specific private key in the libratbag
circle CI organization, meaning that only the master branch of libratbag
will be allowed to push to the doc repo.

Note also that only the master branch will trigger the request:
- any PR will not trigger the request
- any push on the master branch of a personal repo will fail while
  attempting to write the docs, if the repo is setup to use circleCI too.

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@gmail.com>

whot committed:
circle.yml: drop doxygen and graphviz - obsolete now

We're not building docs anymore

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

bentiss commented:
No need to keep this one open
